### DISCUSSION TITLE
How to add doctest / additional checks?

### DISCUSSION THREAD

**Discussion by remexre** [srid/haskell-flake#196] [Q&A]
Possibly an obvious question, but I'm new to flake-parts and haskell-flake. doctest recommends running the tests as `cabal repl --with-ghc=doctest`, rather than as a Cabal test-suite or similar. How do I actually add this to a package's checkPhase?

I'm used to passing `postCheck` in `mkDerivation`'s arguments, but that doesn't seem to be a valid attribute in `haskellProjects.*`.

**Comment by srid** [comment]
See https://github.com/srid/haskell-flake/issues/192

```nix
{
  haskellProjects.default = {
    settings = {
      mypackage = { super, ... }: {
        custom = pkg:  pkg.overrideAttrs (oa: { postCheck = ''..''; });
      };
    };
  };
}
```

We do want to improve this unnecessarily longwinded API though, hence #192.

**Comment by bryceberger** [comment]
For someone who might come across this in the future and wants to run `cabal repl -w doctest` inside the devshell, rather than in the check phase, I found this out:
```nix
{
  haskellProjects.default = {
    # will *not* be able to find any dependencies
    devShell.tools = hs: { inherit (hs) doctest; };
  };
  devShells.default = pkgs.mkShell {
    inputsFrom = [
      config.haskellProjects.default.outputs.devShell
    ];
    nativeBuildInputs = with pkgs; [
      # will find dependencies
      haskellPackages.doctest
    ];
  };
}
```

I am unsure what exactly makes the first one break, as `doctest --version` reports the same GHC in both cases (and in fact both provide the same derivation).
